app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -36,7 +37,27 @@ class EventRepositoryImpl @Inject constructor(
     private val eventPhotoCompressor = EventPhotoCompressor()
 
     override suspend fun upsertEvent(event: Event): UpsertEventResult {
-        val (newEvent, deletedPhotoCount) = compressAndDeletePhotosFromEvent(event)
+        val (compressedPhotos, deletedPhotoCount) = compressAndDeletePhotos(
+            event.photos.filterIsInstance<LocalPhoto>().filter {
+                !it.savedInternally
+            }
+        )
+        val eventWithCompressedPhotos = event.copy(
+            photos = event.photos.filterNot { photo ->
+                photo is LocalPhoto && !photo.savedInternally
+            }.toMutableList().apply {
+                compressedPhotos.forEach { compressedPhoto ->
+                    this.add(compressedPhoto)
+                }
+            }.toImmutableList()
```

Comment:
`toImmutableList()` isn't an official Kotlin list type (you can see in the imports it's some internal OkHttp stuff). You can just use `toList()` here